File: chap5/Puissance.java
// Ces programmes sont sous licence CeCILL-B V1.
class Puissance {

  static int puissance (int n) {
    if (n == 0) {
      return 1;}
    else {
      return (2 * puissance(n-1));}}

  public static void main (String [] args) {
    System.out.println(puissance(10));}}

File: chap5/Puissance.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>

using namespace std;

static int puissance (int n)
{
   if (n == 0) 
   {
      return 1;
   }
   else
   {
      return (2 * puissance(n-1));
   }
}

int main() 
{
   cout << puissance(10) << endl;
}
